src/BotwinExtensions.cs

```diff
@@ -54,6 +59,34 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
             return builder.UseRouter(routeBuilder.Build());
         }
 
+        private static void Apply405Handler(IApplicationBuilder builder, IEnumerable<BotwinModule> modules)
+        {
+            var systemRoutes = new List<(string verb, string route)>();
+            foreach (var module in modules)
+            {
+                foreach (var route in module.Routes.Keys)
+                {
+                    var strippedPath = route.path.EndsWith("/") ? route.path.Substring(0, route.path.Length - 1) : route.path;
+                    systemRoutes.Add((route.verb, "/" + strippedPath));
+                }
+            }
+
+            builder.Use(async (context, next) =>
```

Comment:
If I understand it correctly, then this handler is supposed to “pick up” those routes which paths are registered but are using the incorrect verb? In that case, wouldn’t it be better to run this middleware *after* the normal mapped routes, so it only runs when handling those is over (similarly how in standard ASP.NET, MVC runs *after* StaticFiles *if* StaticFiles could not respond to a request).

Reply:
If it runs after, the routing middleware will set 404 as the status code, the issue is, how do you know its a 404 from an unfound route vs a 404 wrongly assigned for invalid verb on path?

Reply:
Unless I’m completely wrong (routing is one of the topics I didn’t really do anything with yet) but from [how it looks](https://github.com/aspnet/Routing/blob/rel/2.0.0/src/Microsoft.AspNetCore.Routing/RouterMiddleware.cs#L29), the next middleware is only invoked if there is no matching route. So if the middleware is *after* the router, it will only run for unrecognized routes. So you would only have to check whether *any* route with such a path exists in order to determine whether it’s a real not found or just an incorrect verb. So you could do this with a simple hash set on the paths.